gb-dbg/src/dbg_interfaces.rs

```diff
@@ -28,16 +29,20 @@ impl From<u16> for RegisterValue {
     }
 }
 
-pub trait RW {
+pub trait MemoryDebugOperations {
     fn read(&self, index: usize) -> u8;
-
-    fn write(&mut self, _index: usize, _value: u8) {
-        // Default to Read-Only
-    }
 }
 
-pub trait DebugRegister {
-    fn get(&self, key: &str) -> Result<RegisterValue>;
+pub trait RegisterDebugOperations {
+    fn cpu_get(&self, key: &str) -> Result<RegisterValue>;
+
+    fn ppu_get(&self, key: &str) -> Result<RegisterValue>;
+
+    fn io_get(&self, key: &str) -> Result<RegisterValue>;
```

Comment:
Technically PPU registers are part of the IO registers, we should specify in an issue which registers should be displayed as PPU registers and which registers as IO registers.

Reply:
#124